src/main/java/at/hannibal2/skyhanni/features/garden/GardenOptimalSpeed.kt

```diff
@@ -15,18 +16,36 @@ import net.minecraft.client.gui.inventory.GuiEditSign
 import net.minecraftforge.client.event.GuiOpenEvent
 import net.minecraftforge.client.event.GuiScreenEvent.DrawScreenEvent
 import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import kotlin.time.Duration.Companion.milliseconds
 import kotlin.time.Duration.Companion.seconds
 
 class GardenOptimalSpeed {
     private val config get() = GardenAPI.config.optimalSpeeds
     private val configCustomSpeed get() = config.customSpeed
-    private var currentSpeed = 100
+    private var sneakingTime = 0.seconds
+    private val sneaking get() = Minecraft.getMinecraft().thePlayer.isSneaking
+    private val sneakingPersistent get() = sneakingTime > 5.seconds
+    private var _currentSpeed = 100
+    private var currentSpeed: Int
+        get() = (_currentSpeed * (if (sneaking) 0.3 else 1.0)).toInt()
+        set(value) {
+            _currentSpeed = value
+        }
     private var optimalSpeed = -1
     private val currentSpeedPattern = " Speed: §r§f✦(?<speed>.*)".toPattern()
     private var lastWarnTime = 0L
     private var cropInHand: CropType? = null
     private var rancherOverlayList: List<List<Any?>> = emptyList()
 
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (sneaking) {
```

Comment:
Maybe save the time stamp of the last non sneaking moment instead. (Mutable vs immutable state)